src/spec.c

```diff
@@ -1069,21 +1052,69 @@ void IndexSpec_FreeInternals(IndexSpec *spec) {
     }
     rm_free(spec->fields);
   }
-  if (spec->uniqueId) {
-    // If uniqueid is 0, it means the index was not initialized
-    // and is being freed now during an error.
-    IndexSpec_ClearAliases(spec);
+  // ahhhm
+  rm_free(spec->name);
+  // Free sortable list
+  if (spec->sortables) {
+    SortingTable_Free(spec->sortables);
+    spec->sortables = NULL;
   }
 
-  if (spec->keysDict) {
-    dictRelease(spec->keysDict);
-  }
+  rm_free(spec);
+}
 
+void IndexSpec_FreeInternals(IndexSpec *spec) {
```

Comment:
Please explain in a comment what is freed here and what is going to be freed on a background thread, Also maybe call it unlink?

Reply:
I see you have it on function body so forget it, maybe just add a comment explains that this is done on the main thread with the GIL locked and then the rest of the free operation goes to background thread.